docs/matchers/index.rst

```diff
@@ -0,0 +1,388 @@
+Matchers
+==========
+
+Matchers provide a way to validate the arguments passed to :doc:`../mock_callable/index`, :doc:`../mock_async_callable/index` and :doc:`../mock_constructor/index`
+
+Basic Usage
+-----------------------
+
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+  class TestBackupDelete(testslide.TestCase):
+    def setUp(self):
+      super().setUp()
+      self.storage_mock = testslide.StrictMock(storage.Client)
+      # Makes storage.Client(timeout=60) return self.storage_mock
+      self.mock_constructor(storage, 'Client')\
+        .for_call(timeout=matchers.ThisInt(60))\
+        .to_return_value(self.storage_mock)
+    def testrmjson(self)
+      self.mock_callable(os, 'remove')\
+          .for_call(matchers.RegexMatches(".*\.json))\
+          .to_return_value(None)
+
+
+Generic
+-------
+
+Any
+^^^
+testslide.matchers.Any() represents an argument, that anything
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
```

Comment:
I believe this is missing  the class, but I suppose as we have a full example above, we could get by with just mock_callable + matcher lines here.